kubernetes/doc/edge-deployment-to-k8s-translations.md

```diff
@@ -0,0 +1,253 @@
+        - **name** = Name will be the module name, sanitized to be a K8s identifier.
+        - **image** = `settings.image`
+        - **env** = env will contain some predefined settings for all modules, then:
+            - Add all environment variables from `env` section in module spec.
+            - Add all environment variables from `settings.createOptions.Env`.
+        - **securityContext**
+            - privileged = Derived from `settings.createOptions.HostConfig.Privileged`.
+        - **volumeMounts** = There are 4 sources to volume mounts.
+            - bind mounts from `settings.createOptions.HostConfig.Binds` in format "host path:target path[:readwrite mode]".
+              *This is not recommended for IoT Edge on K8s.*
+                - Name = Sanitized host path.
+                - MountPath = target path.
+                - readOnlyProperty = readwrite mode if set, default is false.
+            - bind mounts from `settings.createOptions.HostConfig.Mounts`
+              *This is not recommended for IoT Edge on K8s.*
+                - Name = mount.Source
+                - MountPath = mount.Target
+                - readOnlyProperty = readwrite mode if set, default is false.
+            - volume mounts from `settings.createOptions.HostConfig.Mounts`
+                - Name = mount.Source
+                - MountPath = mount.Target
+                - readOnlyProperty = readwrite mode if set, default is false.
+            - volume mounts from `settings.k8s-extensions.volumes[*].volumeMounts`. Placed in spec 
+              as provided.
+- **imagePullSecrets**
+    If the module server address matches a server address in the edge deployment specification's 
+    *Container Registry Settings*, this will be an array with one name, the created ImagePullSecret.
+- **volumes** There are 4 sources for volumes.
+    - bind mounts from `settings.createOptions.HostConfig.Binds` in format "host path:target path[:readwrite mode]".
+      *This is not recommended for IoT Edge on K8s.*
+        - name = Sanitized host path
+        - hostPath
+            - path = host path.
+            - type = "DirectoryOrCreate".
+    - bind mounts from `settings.createOptions.HostConfig.Mounts`
+      *This is not recommended for IoT Edge on K8s.*
+        - name = mount.Source
+        - hostPath
+            - path = mount.Source
+            - type = "DirectoryOrCreate".
+    - volume mounts from `settings.createOptions.HostConfig.Mounts`
+        - name = mount.Source
+        - persistentVolumeClaim is assigned if edge runtime is started with `persistentVolumeName` 
+          or `storageClassName` set.
+            - claimName = module name + mount.Source
+            - readOnlyProperty = mount.ReadOnly
+        - emptyDir is assigned otherwise.
+    - volume mounts from `settings.k8s-extensions.volumes[*].volume`. Placed in spec as provided.
+- **serviceAccountName** = The module name, sanitized to be a K8s identifier.
```

Comment:
It would be good to explain why serviceaccount is created; and how it is used for customer module.

Reply:
I'll link to the RBAC document, rather than restate it here.